### PR TITLE
Cleanup axis-aligned padding

### PR DESCRIPTION
**Pull request by abarth**
We now have an EdgeInsets.along function that projects the padding onto the
given Axis rather than repeating this function in several places.

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller

File: packages/flutter/lib/src/painting/edge_insets.dart [hunk 1]
```diff
@@ -56,6 +65,18 @@ class EdgeInsets {
   /// The total offset in the horizontal direction.
   double get vertical => top + bottom;
 
+  /// The total offset in the given direction.
+  double along(Axis axis) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/painting/edge_insets.dart]: This is great. There are probably a bunch of classes that would benefit from along(Axis) methods. Maybe Size, Offset, FractionalOffset. No need to do them all right away; this seems like a good pattern to apply in the future.

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/painting/edge_insets.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/painting/edge_insets.dart]: Ah, good idea.

**Pull request merged**